Program.cs

```diff
@@ -58,11 +60,58 @@ private static void ExtractLicense(Product product)
                     if (match.Success)
                     {
                         Console.WriteLine($"Found key for {product.Name}: {match.Captures[0]}");
+                        Console.WriteLine("\nDo you want to save it? (Yes/No)");
+                        string chooseIfSave = "";
+
+                        chooseIfSave = Console.ReadLine();
+
+                        if (chooseIfSave == "Y" || chooseIfSave == "yes" || chooseIfSave == "Yes")
+                        {
+                            TxtFile("Found key for", product.Name, match.Captures[0].ToString());
+                        }
+                        if (chooseIfSave == "N" || chooseIfSave == "no" || chooseIfSave == "No")
+                        {
+                            Console.Clear();
+                        }
                     }
                 }
             }
             catch (Exception) { }
         }
-    }
-}
 
+        private static void ChoosePath(out string path)
+        {
+            path = "";
+            Console.WriteLine("\nPlease type the directory where you want to save the product key: ");
+            path = Console.ReadLine();
+        }
+
+        static private string GenerateFileName(string ProductVS)
+        {
+            string fileName = "";
+            string fileNameExtension = ".txt";
+            string license = "License";
+
+            //Create the name
+            fileName =  ProductVS + " " + license + fileNameExtension;
+
+            return fileName;
+        }
+
+        static public void TxtFile(string header, string ProductVS, string key)
+        {
+            StringBuilder getData = new StringBuilder();
+
+            string fileName = GenerateFileName(ProductVS);
+            string path;
+
+            ChoosePath(out path);
+
+            string fullFilePath = path + "\\" + fileName;
```

Comment:
As a general tip, use Path.Combine() for combining folder names and file names. This will make your code more adapted to running on other operating systems than windows.